### PR TITLE
fix: backup restore events not tracked - WPB-11308

### PR DESCRIPTION
**Pull request by johnxnguyen**
<table>
<td>
  <a href="https://wearezeta.atlassian.net/browse/WPB-11308" title="WPB-11308" target="_blank">BugWPB-11308</a>  [iOS] Backup Restore Succeeded event isn't being logged in Countly
  </td></table>
  <br />
 

### Issue

Backup restore (success and failure) events are not tracked. The events were being sent through the (optional) active user session, but at that point in time there doesn't exist an active user session. The solution is to pass the result of the backup import along to the point where an active user session exists and send the events via the analytics service:

- `BackupRestoreController` passes the result to its delegate (`AuthenticationCoordinator`)
- `AuthenticationCoordinator` enqueues the action `completeBackupStep(didSucceed: Bool?)`
- `AuthenticationCoordinator` executes the action, telling the `UnauthenticatedSession` to report the result. 
- `UnauthenticatedSession` will `upgradeToAuthenticatedSession`, then inform its delegate (`SessionManager`)
- `SessionManager` will create and activate `ZMUserSession`, then send an event to the user session via `userSession.trackAnalyticsEvent(.backupRestored)`.
- If `ZMUserSession` has been configured to send analytics events, it will forward the event to `analyticsEventTracker`.
- Otherwise it add events to `pendingAnalyticsEvents`, and when the `analyticsEventTracker` is set it will forward the pending events.

### Testing

- Log in, restore a backup successfully, enable analytics, check the event is tracked
- Log in, fail to restore a backup (insert wrong password), continue without backup, enable analytics, check the event is tracked

---

### Checklist

- [x] Title contains a reference JIRA issue number like `[WPB-XXX]`.
- [x] Description is filled and free of optional paragraphs.
- [ ] Adds/updates automated tests.

---

### UI accessibility checklist

_If your PR includes UI changes, please utilize this checklist:_
- [ ] Make sure you use the API for UI elements that support large fonts.
- [ ] All colors are taken from WireDesign.ColorTheme or constructed using WireDesign.BaseColorPalette.
- [ ] New UI elements have Accessibility strings for VoiceOver.

### CODE REVIEW

**Review by agisilaos — APPROVED** [review]

File: wire-ios/Wire-iOS/Sources/AnalytitcsServiceConfigurationBuilder.swift [hunk 1]
```diff
@@ -23,14 +23,13 @@ struct AnalyticsServiceConfigurationBuilder {
...
         guard
-            let secretKey = Bundle.countlyAppKey,
             let countlyURL = BackendEnvironment.shared.countlyURL
         else {
             return nil
         }
 
         return AnalyticsServiceConfiguration(
-            secretKey: secretKey,
+            secretKey: "37cc418a6fc07403b0a918252133c15b95486883",
```
Comment (netbe) [inline wire-ios/Wire-iOS/Sources/AnalytitcsServiceConfigurationBuilder.swift]: to revert

File: wire-ios/Wire-iOS/Sources/AnalytitcsServiceConfigurationBuilder.swift [hunk 2]
```diff
@@ -23,14 +23,13 @@ struct AnalyticsServiceConfigurationBuilder {
 
     func build() -> AnalyticsServiceConfiguration? {
         guard
-            let secretKey = Bundle.countlyAppKey,
```
Comment (netbe) [inline wire-ios/Wire-iOS/Sources/AnalytitcsServiceConfigurationBuilder.swift:26]: should stay here, with !secretKey.isEmpty

to test prefer override in debug.xcconfig

**Review by netbe — CHANGES_REQUESTED** [review]
looks good implementation wise, some debug to clean up

File: wire-ios/Wire-iOS/Sources/AnalytitcsServiceConfigurationBuilder.swift (hunk 1 above)
Comment (johnxnguyen) [reply wire-ios/Wire-iOS/Sources/AnalytitcsServiceConfigurationBuilder.swift]: done

File: wire-ios/Wire-iOS/Sources/AnalytitcsServiceConfigurationBuilder.swift (hunk 2 above)
Comment (johnxnguyen) [reply wire-ios/Wire-iOS/Sources/AnalytitcsServiceConfigurationBuilder.swift:26]: done

**Review by netbe — APPROVED** [review]

**Pull request merged**